MAINTAINERS.yml

```diff
@@ -0,0 +1,384 @@
+# This file contains information on who maintains what. It is parsed by
+# get_maintainer.py.
+#
+# File format
+# ###########
+#
+# "Subsystem title" (the quotes are needed if the title has more than one word):
+#     status:
+#         One of the following:
+#
+#         * maintained:
+#             Someone looks after the subsystem.
+#
+#         * odd fixes:
+#             The subsystem has a maintainer, but they don't have time to do
+#             much other than throw the odd patch in.
+#
+#         * orphaned:
+#             No current maintainer (but maybe you could take the role as you
+#             write your new code).
+#
+#         * obsolete:
+#             Old code. Something being marked obsolete generally means it has
+#             been replaced by a better subsystem that you should be using
+#             instead.
+#
+#     maintainers:
+#         List of GitHub handles for the people who maintain the subsystem.
+#         Usually, there's only one maintainer.
+#
+#     collaborators (not to be confused with the GitHub collaborator role):
+#         Very involved contributors, who know the area well and contribute
+#         significantly to it.
+#
+#     inform:
+#         List of GitHub handles for people to inform when a pull request that
+#         modifies the subsystem is sent.
+#
+#     labels:
+#         List of GitHub labels to add to pull requests that modify the
+#         subsystem.
+#
+#     files:
+#         List of paths and/or glob patterns giving the files in the subsystem,
+#         relative to the root directory.
+#
+#         If a path or glob pattern ends in a '/', it matches all files within
+#         the given directory or directories. Otherwise, an exact match is
+#         required.
+#
+#         Paths to directories should always have a trailing '/'.
+#
+#     files-regex:
+#         List of regular expressions applied to paths to determine if they
+#         belong to the subsystem. The regular expression may match anywhere
+#         within the path, but can be anchored with ^ and $ as usual.
+#
+#         Can be combined with a 'files' key.
+#
+#         Note: Prefer plain 'files' patterns where possible. get_maintainer.py
+#         will check that they match some file, but won't check regexes
+#         (because it might be slow).
+#
+#     files-exclude:
+#         Like 'files', but any matching files will be excluded from the
+#         subsystem.
+#
+#     files-regex-exclude:
+#         Like 'files-regex', but any matching files will be excluded from the
+#         subsystem.
+#
+#     description: >-
+#         Plain-English description. Describe what the system is about, from an
+#         outsider's perspective.
+#
```

Comment:
add a new field for category, for example:
- driver
- board
- soc 
- architecture
- middleware
- ..

This way we can make better judgment who the assignee (the person driving a Pr to a mergeable state)

Reply:
Don't know how it'd get used, so need some feedback:

 - Should it be mandatory or optional?

 - Should it be a freeform string, or is there a list of acceptable values?

Reply:
Would it make sense to add this as a further PR after? As it seems it is not too clear yet

Reply:
> Would it make sense to add this as a further PR after? As it seems it is not too clear yet

why? once this is merged, it means the syntax is approved, we should not merge this until we have agreed on the structure and syntax. There is no use for this being merged now anyways.

> As it seems it is not too clear yet

Here is the usecase: you submit a driver that is associated with a board, you have 2 owners/maintainers out of that, you can then make a decision to assign to the board owner instead of assigning to the driver, just an example.
Having this hierarchy and knowledge about what are the code changes are (changing a driver vs changing an architecture) can be easily scriptable.

Reply:
> once this is merged, it means the syntax is approved, we

We can always add new fields after.

Anyhow, @nashif , if you would want it now, I guess Ulf would need a bit more of detail to know what to implement: https://github.com/zephyrproject-rtos/zephyr/pull/20591#discussion_r345375329

Reply:
> We can always add new fields after

Yeah, I'd prefer to keep things minimal to start with. Never too late to add stuff.

> Anyhow, @nashif , if you would want it now, I guess Ulf would need a bit more of detail to know what to implement: [#20591 (comment)](https://github.com/zephyrproject-rtos/zephyr/pull/20591#discussion_r345375329)

Yeah, will need more information on `category` at least (see above).